geps/gep-1619/index.md

```diff
@@ -3,22 +3,21 @@
 * Issue: [#1619](https://github.com/kubernetes-sigs/gateway-api/issues/1619)
```

Comment:
Can't comment directly on the title, but given the rename of the API, we _may_ also want to rename the GEP to match.